torch_geometric/datasets/metrla.py

```diff
@@ -0,0 +1,177 @@
+    if :math:`dist(v_{i}, v_{j}) >= K`, otherwise :math:`0`, where K is a
+    pre-fixed threshold, and dist(, ) is the real-world road distance
+    between two sensors, and :math:`\sigma^{2}` is the variance of the
+    distances between sensors. A zero-value in the adjacency matrix does
+    not indicate that thers is no road connection between two sensors (
+    because there almost always is, since road networks are connected
+    graphs. Instead, it means that the distance between them is larger than
+    the threshold value.
+
+    Args:
+        root (string): Root directory where the dataset should besaved.
+        n_previous_steps (int): The number of previous time steps to consider
+            when building the predictor variable.
+        n_future_steps (int): The number of next time steps to consdier when
+            building the target variable.
+        add_time_of_day (bool): Whether to inject day of week information in
+            the features.
+        add_day_of_week (bool): Whether to inject time of day information in
+            the features.
+        normalized_k (float):  The threshold for constructing the adjacency
+            matrix based on the thresholded Gaussian kernel.
+        transform: A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+        pre_transform: A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before being saved to
+            disk
+    """
+    gdrive_ids = {
+        'distances.csv': '19Td6JafGnF8CD2H64jWCBV7k5GyFix2H',
+        'locations.csv': '1FnioVF2jZuOl_St1ssLnvQgSHuEuDzvL',
+        'sensors.txt': '1235bXBxe4X73dJk3zwRzaIQ-8A7K_OEV',
+        'sensor-readings.h5': '1VgLjs4XB5O-IpcyZ5GAQG9sGuyxKe3Lc'
+    }
+
+    def __init__(self, root: Optional[str], n_previous_steps: int,
+                 n_future_steps: int, add_time_of_day: bool = False,
+                 add_day_of_week: bool = False, normalized_k: float = .1,
+                 transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None) -> None:
+        """This is a circular dependency. You need this to calculate processed
+        file names, but you need it before the call call to the "process"
+        method in the superclass. The only in-between place you could
+        effectively calculate this is before the process, so either in
+        __init__ (where you don't have access to the downloaded file
+        containing the readings, or in download, but download gets skipped
+        after the first time, so that's not reliable. Open to any
+        workarounds, but the dataset has a fixed size, so
+        this is not a deal breaker imo."""
+        self.n_readings = 34272
```

Comment:
It's probably sufficient to just test that the first few processed files exists. Otherwise, checking the existence of 34272 can quickly become the bottleneck.

Reply:
Checking the existence of ~34K file takes 1.2 seconds in my case (on SSD; HDD storage might be slower indeed), so it's probably not that much of a bottleneck for something that happens every time the code is run (if I remove the test case).

Anyway, your suggestion is valid, and we should optimize wherever possible. I modified it to check only the first 5 (needed a bit more changes in the code than replacing the value with 5, i.e. in the "len" method), and it's coming with the next push. Thanks!